### PR TITLE
Display graph

### PR DESCRIPTION
**Pull request by evulins**
Done: 
- displaying graph,
- displaying scrollbar,
- displaying tooltip,
- displaying custom arrows for navigation,
- while using scrollbar, we are also changing values in Stream info.

Scrollbar is not styled as it is in the designs.

[Trello card](https://trello.com/c/dvX1cq18/1828-session-details-graph)

### CODE REVIEW

File: app/javascript/react/assets/styles/global-styles.ts [hunk 1]
```diff
@@ -36,6 +36,9 @@ const GlobalStyles = createGlobalStyle`
     line-height: 1.2;
     color: ${gray400};
   }
+
+
+  .gm-style iframe + div { border:transparent  !important; }
```
Comment (smialko) [inline app/javascript/react/assets/styles/global-styles.ts]: too many spaces here: `transparent  !important;` and additional empty line :D

File: app/javascript/react/components/Graph/Graph.tsx [hunk 2]
```diff
@@ -28,6 +32,17 @@ const Graph = (props: HighchartsReact.Props) => {
...
+  const onMouseOverSingle = (latLng: {
+    latitude: number;
+    longitude: number;
+  }) => {
+    console.log(latLng);
+  };
+
+  const onMouseOverMultiple = (xLess: number, xMore: number) => {
+    console.log(xLess, xMore);
+  };
```
Comment (smialko) [inline app/javascript/react/components/Graph/Graph.tsx]: Do we want it to stay?

File: app/javascript/react/components/Graph/Graph.tsx [hunk 3]
```diff
@@ -28,6 +32,17 @@ const Graph = (props: HighchartsReact.Props) => {
 
   const yAxisOption = getYAxisOption(thresholdsState);
 
+  const onMouseOverSingle = (latLng: {
+    latitude: number;
+    longitude: number;
+  }) => {
+    console.log(latLng);
```
Comment (smialko) [inline app/javascript/react/components/Graph/Graph.tsx]: and this?

File: app/javascript/react/assets/styles/global-styles.ts [hunk 4]
```diff
@@ -36,6 +36,10 @@ const GlobalStyles = createGlobalStyle`
     line-height: 1.2;
     color: ${gray400};
   }
+
+//   .highcharts-scrollbar {
+//     top: 0;
+// }
 `;
```
Comment (smialko) [inline app/javascript/react/assets/styles/global-styles.ts:43]: leftoover

**Review by smialko — APPROVED** [review]
Looking good, working great. Excellent job! 👏

File: app/javascript/react/components/Graph/Graph.tsx (hunk 2 above)
Comment (evulins) [reply app/javascript/react/components/Graph/Graph.tsx]: no, but i think this is from older commit

File: app/javascript/react/components/Graph/Graph.tsx (hunk 3 above)
Comment (evulins) [reply app/javascript/react/components/Graph/Graph.tsx]: Same here

File: app/javascript/react/assets/styles/colors.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ const gray100 = "#F4F6F9";
 const gray200 = "#C0C9D1";
 const gray300 = "#7D858C";
 const gray400 = "#3E4449";
-const gray500 = '#C6CDD3';
+const gray500 = "#C6CDD3";
```
Comment (jankrzempek) [inline app/javascript/react/assets/styles/colors.ts:9]: 👮🏻

File: app/javascript/react/components/Graph/Graph.tsx [hunk 6]
```diff
@@ -1,62 +1,137 @@
...
-  seriesOption,
+  seriesOptions,
+  getYAxisOptions,
+  responsive,
+  getTooltipOptions,
+  scrollbarOptions,
+  credits,
+  getRangeSelectorOptions,
 } from "./graphConfig";
+import { useSelector } from "react-redux";
```
Comment (jankrzempek) [inline app/javascript/react/components/Graph/Graph.tsx:18]: external import to those above please

File: app/javascript/react/components/Graph/Graph.tsx [hunk 7]
```diff
@@ -1,62 +1,137 @@
...
+  const yAxisOption = getYAxisOptions(thresholdsState);
+  const tooltipOptions = getTooltipOptions(measurementType, unitSymbol);
+  const rangeSelectorOptions = getRangeSelectorOptions(
+    fixedSessionTypeSelected
+  );
+
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    if (measurements.length > 0 && !isLoading) {
```
Comment (jankrzempek) [inline app/javascript/react/components/Graph/Graph.tsx:79]: isn't it enough to state `measurements.length && !isLoading`?

File: app/javascript/react/components/Graph/Graph.tsx [hunk 8]
```diff
@@ -1,62 +1,137 @@
...
+      const now = Date.now();
+      const last24Hours = measurements.filter(
+        (m) => now - m.time <= MILLISECONDS_IN_A_DAY
+      );
+      if (last24Hours.length > 0) {
```
Comment (jankrzempek) [inline app/javascript/react/components/Graph/Graph.tsx:84]: here the same, do we need > 0?

File: app/javascript/react/components/Map/Map.tsx [hunk 9]
```diff
@@ -93,59 +122,126 @@ const Map = () => {
...
+        : dispatch(fetchMobileStreamById(streamId));
+    }
+
+    if (!selectedStreamId) {
+      setSelectedSessionId(id);
+      setSelectedStreamId(streamId);
+      setModalOpen(false);
+      setTimeout(() => {
+        setModalOpen(true);
+      }, 0);
```
Comment (jankrzempek) [inline app/javascript/react/components/Map/Map.tsx:143]: if the timeout is 0 can we just use `setModalOpen(true)`?

**Review by jankrzempek — APPROVED** [review]
WOW! Well crafted 🙌🏻

One thing, left arrow is cut off a bit 
Zrzut ekranu 2024-06-13 o 19 41 34

File: app/javascript/react/components/Map/Map.tsx (hunk 9 above)
Comment (evulins) [reply app/javascript/react/components/Map/Map.tsx:143]: This is still valid solution even with 0  -> https://medium.com/@madhavkhera868/article-on-zero-delay-settimeout-function-in-javascript-f25a08ca350b

**Pull request merged**